### DISCUSSION TITLE
Changing object detection model's training data

### DISCUSSION THREAD

**Discussion by klutzzykitty** [blakeblackshear/frigate#4582] [Ask A Question]
Firstly, just to avoid misunderstandings, this isn't related to custom models discussion on #1043 
but aligns more towards understanding Frigate's AI better as discussed on #4470 

I understand frigate plus is featured to have custom models tuned to people's requirements. 

Does the open source version of Frigate throw light on how the user can train the default model for better accuracy on the data that 
it is already trained on ? 

As the default _Mobiledet_ model comes pre-trained on COCO, the drawbacks of COCO are also carried forward to the end user 
where it works good for common objects in outdoor environments such as vehicles, people, animals etc but also has redundant objects that might never get used , like toasters, books, bowls, toothbrushes etc. 

Some of these can be replaced with more important objects such as weapons, fire etc which after training can be updated as the default model instead of the plain COCO traind existing one. 

I understand discussions close to this have happened before on #1237 and are happening on #1043  as well. 
But with the intention to contribute towards the AI part of Frigate, I can easily rig up a change for the default mobiledet model with at least one important object included, _provided, some info on training the existing model can be shared._ 

Thereby, it wouldn't affect the custom models track as that's more of a business add on with Frigate plus. 

Thanks !!

**Comment by blakeblackshear** [comment]
Frigate can run any object detection model designed for tflite and/or the Coral: https://coral.ai/docs/edgetpu/models-intro/

You can see another open source effort as well: https://gitlab.com/Curid/TF-CCTV

**Reply by klutzzykitty** [reply]
Thanks for the reply Blake!

Yeah, I know that Frigate runs whatever object detection model that's compiled for edge TPU. 
I've working on TF, TFlite and coral API before. 

My concern is that the low latency model _mobiledet_ is built using TF1 and Google has nuked TF1 support for python 3.8 and later.
Coral on the other hand haven't updated their workflow for training _mobiledets_ (their present workflow to retrain any edgetpu model, doesn't work ever since TF1 support was removed. Only the TF2 workflows hold good, which isn't suitable to be used with Frigate as its latency is 37-40ms compared to the one Frigate uses - _mobiledet_  which is at 9-10ms).

So, If Frigate's custom models are being trained, there has to be a working workflow to retrain the model on different datasets isn't it ? 

And if such a retraining method exists with Frigate, has it been shared ?
Or has Frigate used the same TF1 methodology to train on custom data?

**Reply by blakeblackshear** [reply]
I see. Yea. Colab doesn't work anymore, but most of what you need is still there. You need a GPU instance where you have more control. I would look at the TF CCTV repo again as it has most of what you want to know and is somewhat similar to what I am doing in terms of the training process. For now, I am keeping my exact approach a bit of a trade secret because it wasn't easy.

**Reply by klutzzykitty** [reply]
Ah! yeah , makes sense. 

I'm also trying to come up with a reliable approach. 
Thanks for pointing towards TF-CCTV.

**Reply by StefanoGiu** [reply]
Hello, I'm also facing issue with the standard TPU model of Frigate... it's not reliable at all... My dog is very often detected as a person, and the same my lawnmower robot... so I cannot use it for videosurveillance with my CCTV IP cameras.

How is the TF-CCTV model ? Did you improve the detection reliability?

**Comment by StefanoGiu** [comment]
Hello, I'm also facing issue with the standard TPU model of Frigate... it's not reliable at all... My dog is very often detected as a person, and the same my lawnmower robot... so I cannot use it for videosurveillance with my CCTV IP cameras.

How is the TF-CCTV model ? Did you improve the detection reliability?

**Reply by klutzzykitty** [reply]
It only detects "person" class.

**Reply by StefanoGiu** [reply]
Only person is good!! Is it reliable? What about the comparison with the standard frigate model? Do you have false positives? Precision/Recall?

**Reply by klutzzykitty** [reply]
Not precisely sure. A good comparison has to be done. 
Also, standard frigate model is the google coral example model. Not really a frigate issue as the example model can only do so much. 
Yeah false positives are indeed present but you can say its a bit less. 

Frigate gives excellent methods to counter false positives with class filter scores. 
Set the person detection score to a high value in config

**Comment by StefanoGiu** [comment]
My dog is recognized as person with score of 85% or more... so this will not help... Same happens for shadows of the tree/leaves made by the sun... it's terrible...
